File: mobile/lib/features/auth/infrastructure/auth_api.dart
// =====================================================================
// KineticOs — Cliente REST del módulo auth (/api/v1/auth).
// =====================================================================
import 'package:dio/dio.dart';

import '../domain/auth_user.dart';

class AuthResult {
  final String accessToken;
  final String refreshToken;
  final AuthUser user;

  const AuthResult({required this.accessToken, required this.refreshToken, required this.user});

  factory AuthResult.fromJson(Map<String, dynamic> json) {
    return AuthResult(
      accessToken: json['accessToken'] as String,
      refreshToken: json['refreshToken'] as String,
      user: AuthUser.fromJson(json['user'] as Map<String, dynamic>),
    );
  }
}

class AuthApi {
  final Dio _dio;

  AuthApi(this._dio);

  Future<AuthResult> register(String email, String password) async {
    final response = await _dio.post('/auth/register', data: {'email': email, 'password': password});
    return AuthResult.fromJson(response.data as Map<String, dynamic>);
  }

  Future<AuthResult> login(String email, String password) async {
    final response = await _dio.post('/auth/login', data: {'email': email, 'password': password});
    return AuthResult.fromJson(response.data as Map<String, dynamic>);
  }

  Future<AuthUser> me() async {
    final response = await _dio.get('/auth/me');
    return AuthUser.fromJson(response.data as Map<String, dynamic>);
  }
}

File: web/src/features/auth/api/authApi.ts
// =====================================================================
// MiCoach — Llamadas HTTP del módulo auth (/api/v1/auth).
// =====================================================================
import { apiClient } from '@/core/api/client';
import type { AuthResponse, AuthUser } from '../domain/authTypes';

export const authApi = {
  register(email: string, password: string) {
    return apiClient.post<AuthResponse>('/auth/register', { email, password }).then((r) => r.data);
  },

  login(email: string, password: string) {
    return apiClient.post<AuthResponse>('/auth/login', { email, password }).then((r) => r.data);
  },

  me() {
    return apiClient.get<AuthUser>('/auth/me').then((r) => r.data);
  },
};
